app/src/main/java/com/col/eventradar/utils/MapUtils.kt

```diff
@@ -251,22 +281,49 @@ object MapUtils {
         searchResult: LocationSearchResult,
         toastFragment: ToastFragment,
         binding: FragmentMapBinding,
+        countries: List<String>
     ) {
         try {
             val result = OpenStreetMapService.api.getLocationDetails(placeId = searchResult.placeId)
             val feature = toMapLibreFeature(result)
 
-            map.style?.getSource(SEARCH_RESULT_AREA_SOURCE_NAME)?.apply {
-                if (this is GeoJsonSource) {
-                    setGeoJson(FeatureCollection.fromFeature(feature))
+            binding.apply {
+                if (!countries.contains(feature.getStringProperty("placeId"))) {
+                map.style?.getSource(SEARCH_RESULT_AREA_SOURCE_NAME)?.apply {
+                    if (this is GeoJsonSource) {
+                        setGeoJson(FeatureCollection.fromFeature(feature))
+                    }
                 }
-            }
 
-            binding.apply {
                 mapAddLocationButton.visibility = View.VISIBLE
                 mapAddLocationButton.setOnClickListener {
+
                     toastFragment("Added ${feature.getStringProperty("localname")} to User")
                     mapAddLocationButton.visibility = View.GONE
+
+                    val userAreaManager =
+                        UserAreaManager(userRepository = UserRepository(binding.root.context))
+                    val areaRepository = AreasOfInterestRepository(binding.root.context)
+                    FirebaseAuth.getInstance().addAuthStateListener { auth ->
+                        coroutineScope.launch {
+                            if (auth.currentUser != null) {
+                                val currentUserId = auth.currentUser!!.uid
+                                if (userAreaManager.getUser(currentUserId)?.areasOfInterest?.all {it.placeId != feature.getStringProperty("placeId")} != false){
```

Comment:
```suggestion
                                if ((userViewModel || userRepo).getUser(currentUserId)?.areasOfInterest?.all {it.placeId != feature.getStringProperty("placeId")} != false){
```
dont use `userAreaManager` to get data about user that already can be got from the userRepo. if you use this function in a viewmodel it makes sense. maybe check even in general if the `userAreaManager` is needed if you move the function else where (viewmodel)